ArchiSteamFarm/Steam/Integration/SteamChatMessage.cs

```diff
@@ -0,0 +1,193 @@
+// You may obtain a copy of the License at
+// |
+// http://www.apache.org/licenses/LICENSE-2.0
+// |
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if NETFRAMEWORK
+using ArchiSteamFarm.Compatibility;
+#endif
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace ArchiSteamFarm.Steam.Integration {
+	internal static class SteamChatMessage {
+		internal const char ContinuationCharacter = '…'; // A character used for indicating that the next newline part is a continuation of the previous line
+		internal const byte ContinuationCharacterBytes = 3; // The continuation character specified above uses 3 bytes in UTF-8
+		internal const ushort MaxMessageBytes = 6449; // This is a limitation enforced by Steam
+		internal const ushort MaxMessagePrefixBytes = MaxMessageBytes - ReservedContinuationMessageBytes - ReservedEscapeMessageBytes; // Simplified calculation, nobody should be using prefixes even close to that anyway
+		internal const byte NewlineWeight = 61; // This defines how much weight a newline character is adding to the output, limitation enforced by Steam
+		internal const byte ReservedContinuationMessageBytes = ContinuationCharacterBytes * 2; // Up to 2 optional continuation characters
+
+		private const byte ReservedEscapeMessageBytes = 5; // 2 characters total, escape one '\' of 1 byte and real one of up to 4 bytes
+
+		internal static string Escape(string message) {
+			if (string.IsNullOrEmpty(message)) {
+				throw new ArgumentNullException(nameof(message));
+			}
+
+			return message.Replace("\\", "\\\\", StringComparison.Ordinal).Replace("[", "\\[", StringComparison.Ordinal);
+		}
+
+		internal static async IAsyncEnumerable<string> GetMessageParts(string message, string? steamMessagePrefix = null) {
+			if (string.IsNullOrEmpty(message)) {
+				throw new ArgumentNullException(nameof(message));
+			}
+
+			int prefixBytes = 0;
+
+			if (!string.IsNullOrEmpty(steamMessagePrefix)) {
+				// We must escape our message prefix if needed
+				steamMessagePrefix = Escape(steamMessagePrefix!);
+
+				string[] prefixLines = steamMessagePrefix.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);
+
+				prefixBytes = prefixLines.Where(prefixLine => prefixLine.Length > 0).Sum(Encoding.UTF8.GetByteCount) + ((prefixLines.Length - 1) * NewlineWeight);
+
+				if (prefixBytes > MaxMessagePrefixBytes) {
+					throw new ArgumentOutOfRangeException(nameof(steamMessagePrefix));
+				}
+			}
+
+			const int maxMessageBytes = MaxMessageBytes - ReservedContinuationMessageBytes;
+
+			// We must escape our message prior to sending it
+			message = Escape(message);
+
+			int bytesRead = 0;
```

Comment:
Consider renaming this variable to `bytesInMessage` (or `bytesInMessageForSteam`, because newline is considered for more bytes than it actually is), since it doesn't reflect bytes count we have actually read.

Reply:
This is alright, done.